0-limit-crawler/main.go

```diff
@@ -35,15 +37,33 @@ func Crawl(url string, depth int, wg *sync.WaitGroup) {
 	for _, u := range urls {
 		// Do not remove the `go` keyword, as Crawl() must be
 		// called concurrently
-		go Crawl(u, depth-1, wg)
+		go func() {
+			<-*allowCrawlChannel
+			fmt.Println(fmt.Sprintf("Crawling url %s at timestamp %s", url, time.Now()))
+			Crawl(u, depth-1, wg, allowCrawlChannel)
+		}()
 	}
 	return
 }
 
 func main() {
 	var wg sync.WaitGroup
+	allowCrawlChannel := make(chan interface{})
+	ticker := time.NewTicker(1 * time.Second)
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				break
+			case <-ticker.C:
+				allowCrawlChannel <- nil
+			}
+		}
+	}()
 
 	wg.Add(1)
-	Crawl("http://golang.org/", 4, &wg)
+	Crawl("http://golang.org/", 4, &wg, &allowCrawlChannel)
```

Comment:
Have you considered passing down the ticker directly? What's the advantage of creating another channel?